### PR TITLE
Prepare plugin for Marketplace

### PR DESCRIPTION
**Pull request by sgiehl**
In order to publish the plugin to the Marketplace as of Matomo 4, there are a few adjustments required.
Additionally some parts from core need to be moved to this plugin, so core works without it.

Note: The current 4.x-dev branch includes the latest files from 4.x-dev branch of matomo repo. (The history was transfered using git filter-branch). 

refs https://github.com/matomo-org/matomo/pull/16090

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+env:
+  global:
+    - PLUGIN_NAME=CustomVariables
+    - PIWIK_ROOT_DIR=$TRAVIS_BUILD_DIR/matomo
+    # this variable controls the version of Matomo your tests will run against.
+    # by default it will run against the maximum support version read from plugin.json
+    # (PIWIK_TEST_TARGET=maximum_supported_piwik).
+    # You can also specify a specific Matomo version
+    # (PIWIK_TEST_TARGET=2.16.0-b1).
+    - PIWIK_TEST_TARGET=11524-cvar
```
Comment (sgiehl) [inline .travis.yml:34]: Note: That needs to be changed to 4.x-dev once the PR in matomo is merged

**Comment by sgiehl** [comment]
Before publishing to the marketplace we should also add a README and maybe some screenshots

**Comment by sgiehl** [comment]
@tsteur @diosmosis  could someone do a quick review. I've created #2 as follow up for the changes we may need to apply before releasing the plugin on the marketplace...

**Comment by tsteur** [comment]
@sgiehl did some tests and worked fine for me. Tracked custom variables for various slots in various scopes. It's shown in visitor log and the reports. The manage page works as well 👍

**Pull request merged**